`make render-docs` broken due to permissions issue

Observed on Cilium master today:

```
$ echo $UID
1000
$ sudo make -C Documentation clean
make: Entering directory '/home/joe/work/src/github.com/cilium/cilium/Documentation'
rm -rf _build
make: Leaving directory '/home/joe/work/src/github.com/cilium/cilium/Documentation'
```
```
$ make render-docs
Sending build context to Docker daemon  10.24kB
Step 1/7 : FROM docker.io/library/python:3.7.5-alpine3.10
 ---> 8922d588eec6
Step 2/7 : LABEL maintainer="maintainer@cilium.io"
 ---> Using cache
 ---> 8827c6c44539
Step 3/7 : RUN apk add --no-cache --virtual --update     aspell-en     bash     ca-certificates     enchant     git     libc6-compat     py-pip     python     sphinx-python     && true
 ---> Using cache
 ---> 0c7de3b2c7a4
Step 4/7 : ADD ./requirements.txt /tmp/requirements.txt
 ---> Using cache
 ---> ff38ed6d2edc
Step 5/7 : RUN pip install -r /tmp/requirements.txt
 ---> Using cache
 ---> 3c53fe2dfc0a
Step 6/7 : ENV HOME=/tmp
 ---> Using cache
 ---> 0cdf191a4596
Step 7/7 : RUN install -m 0777 -d /usr/local/lib/python3.7/site-packages/versionwarning/_static/data
 ---> Using cache
 ---> 8d10339d5cd3
Successfully built 8d10339d5cd3
Successfully tagged cilium/docs-builder:latest
  GEN   Documentation/_api
  GEN   Documentation/_build/html
docs-cilium
9c760c89fb47551409f69a0dbd40676b84ed06371eb8850ce7a89e40a98a57e1
Validating documentation (syntax, spelling)...
./check-build.sh: line 24: /src/Documentation/_build/warnings.txt: Permission denied
Building documentation (html)...
Running at http://localhost:9081
/usr/local/lib/python3.7/site-packages/sphinx/application.py:402: RemovedInSphinx20Warning: app.info() is now deprecated. Use sphinx.util.logging instead.
  RemovedInSphinx20Warning)

Exception occurred:
  File "/usr/local/lib/python3.7/os.py", line 221, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/src/Documentation/_build/doctrees'
The full traceback has been saved in /tmp/sphinx-err-3og3fuid.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make[1]: *** [Makefile:37: html] Error 2
make: *** [Makefile:461: render-docs] Error 2
```

```
$ ls -lh Documentation/_build
total 4.0K
drwxr-xr-x 2 root root 4.0K Apr  6 17:10 html
```
```
$ make -C Documentation/ clean
rm: cannot remove '_build/html': Permission denied
```

--- a/Documentation/Makefile
+++ b/Documentation/Makefile
@@ -39,6 +39,9 @@ epub latex html: builder-image
 DOCS_PORT = 9081
 
 run-server: stop-server
+	# Work around Docker issue where this directory is created as root in
+	# the `--volume` parameter below. See also GH-10869.
+	$(QUIET)mkdir -p $(CURDIR)/_build/html/
 	$(QUIET)docker container run --rm \
 		--detach \
 		--interactive \

